src/integration/inference/rerank.test.ts

```diff
@@ -34,4 +34,99 @@ describe('Integration Test: Pinecone Inference API rerank endpoint', () => {
     // (Just ignoring the fact that technically doc.document['text'] could be undefined)
     expect(response.data.map((doc) => doc.document['text'])).toBeDefined();
   });
+
+  test('Confirm list of strings as docs + rankFields set to customField fails', async () => {
+    const myDocuments = ['doc1', 'doc2'];
+    const rankFields = ['customField'];
+
+    await expect(
+      pinecone.inference.rerank(model, query, myDocuments, { rankFields })
+    ).rejects.toThrow(
+      expect.objectContaining({
+        message: expect.stringContaining(
+          "Field 'customField' not found in document at index 0"
+        ),
+      })
+    );
+  });
+
+  test('Confirm docs as list of objects + no rankFields succeeds, if docs contain `text` key, succeeds', async () => {
+    const myDocuments = [{ text: 'doc1' }, { text: 'doc2' }];
+    expect(pinecone.inference.rerank(model, query, myDocuments)).toBeDefined();
+  });
+
+  test('Confirm docs as list of objects with additional customField + no rankfields, succeeds', async () => {
+    const myDocuments = [
+      { text: 'hi', customField: 'doc1' },
+      { text: 'bye', customField: 'doc2' },
+    ];
+    expect(pinecone.inference.rerank(model, query, myDocuments)).toBeDefined();
```

Comment:
I'm a little rusty on ts at this point, if the call fails it would avoid the expect and fail the test, right? Just trying to understand what this is asserting.

Reply:
oh man, extremely good catch! I'll change to assert that rerankUnits are >= 1